packages/api/endpoints/async-operations.js

```diff
@@ -42,7 +46,35 @@ async function getAsyncOperation(req, res) {
   return res.send(pick(asyncOperation, ['id', 'status', 'taskArn', 'description', 'operationType', 'output']));
 }
 
+/**
+ * Delete an async operation
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function deleteAsyncOperation(req, res) {
+  const {
+    asyncOperationModel = new AsyncOperationModel({
+      stackName: process.env.stackName,
+      systemBucket: process.env.system_bucket,
+      tableName: process.env.AsyncOperationsTable,
+    }),
+    asyncOperationPgModel = new AsyncOperationPgModel(),
+    knex = await getKnexClient(),
+  } = req.testContext || {};
+
+  const { id } = req.params;
+
+  await knex.transaction(async (trx) => {
+    await asyncOperationPgModel.delete(trx, { id });
+    await asyncOperationModel.delete({ id });
```

Comment:
Nit (and after writing it, probably out of scope for this PR): Minor concern that downstream changes/api code might inadvertantly result in the `.delete` call getting something like 

`{ id: undefined }` (which is what happens if 'id' isn't present in req.params) and having bad behavior.   It's not covered in the database model units, and we're not doing input validation here as the route ensures the param is present.    Could/should we explicitly throw in the endpoint method if { id }  doesn't exist, or add a base model unit that explicitly prevents `d.delete(knex, {})` in favor of `d.deleteAll(knex)`

Reply:
So I'm trying to add behavior to catch where `id` is undefined for this endpoint and return a 400, but I don't think that can happen? If you do `DELETE /asyncOperations/` (excluding the id parameter), then you get a 404 because it's not a valid endpoint?

This unit test I'm writing fails because it's returning 404 and not even getting to my handler to return the 400

```
  const response = await request(app)
    .delete('/asyncOperations/')
    .set('Accept', 'application/json')
    .set('Authorization', `Bearer ${jwtAuthToken}`)
    .expect(400);

  t.is(response.status, 400);
```

So I'm not sure if there's anything we need to do here?

Reply:
Right, that's what I meant by: 

`and we're not doing input validation here as the route ensures the param is present.` 

It's fair that a unit test using the router can't create that scenario *as currently written*. 

I'm low-key concerned about their being the possibility of a future change resulting in that being possible and as the model has no defense against it resulting in an inadvertent table wipe.     It's out of scope for this PR, but perhaps something we should consider addressing.  I can add it as a brief discussion in the arch meeting.